app/controllers/api/v0/highlights_controller.rb

```diff
@@ -10,12 +10,14 @@ def create
     inbound_binding, error = bind(params.require(:highlight), Api::V0::Bindings::NewHighlight)
     render(json: error, status: error.status_code) and return if error
 
-    created_highlight = service_limits.with_create_protection do |user|
-      inbound_binding.create_model!(user_id: user.id)
+    source_id = params[:highlight][:source_id]
+    Highlight.with_advisory_lock("#{current_user_uuid}#{source_id}") do
```

Comment:
`source_id` may not be downcased here which could lead to incorrect advisory lock names (since the downcasing happens in the model).

Can you add a private method to clean up the advisory lock call here and below?

```ruby
def with_advisory_lock(something_with_source_id)
  Highlight.with_advisory_lock("#{current_user_uuid}#{something_with_source_id.source_id}") do
    yield
  end
end
```

and then this becomes

```ruby
with_advisory_lock(inbound_binding) do
  ...
end
```

And down in `destroy` you'll have

```ruby
with_advisory_lock(@highlight) do
  ...
end
```

Reply:
also this private method can duplicate the `source_id` normalization present in the model to make sure it has been applied.